### ISSUE TITLE
Versioning of email templates

### ISSUE THREAD

**Issue by davidmw** [FusionAuth/fusionauth-issues#143]
### Problem
Any given email template can be expected to evolve over time. There might be obsolete versions, an active version, and a version currently in development / pending approval status.

The Welcome / Account verification emails likely need content beyond the basics of what is needed by the FusionAuth end of things. Instructions, URLs and the like, plus css styles and formatting will all need to be refined and corrected over time.

### Solution
A versioning system for each email template would be ideal. Each template would probably have a document id and support one active version. Other time stamped versions would have status indicating things like 'retired' and 'pending.'

### How to vote
Please give us a thumbs up or thumbs down as a reaction to help us prioritize this feature. Feel free to comment if you have a particular need or comment on how this feature should work.

**Comment by soullivaneuh** [comment]
A better idea would to provide template files support (like #430).

With that, versioning can be done with ease with a versioning tool like git.

**Comment by davidmw** [comment]
@soullivaneuh Not really. I agree that improving deployment of templates is needed. I've edited this ticket since the focus wasn't intended to be on specific versioning techniques but rather on surfacing email versioning in the FA UI.
